pkg/controller/certificatemanager/certificatemanager.go

```diff
@@ -0,0 +1,286 @@
+package certificatemanager
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/openshift/library-go/pkg/crypto"
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/controller/status"
+	"github.com/tigera/operator/pkg/controller/utils/imageset"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/tls"
+	"github.com/tigera/operator/pkg/tls/certificatemanagement"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var (
+	log                    = logf.Log.WithName("tls")
+	ErrInvalidCertDNSNames = errors.New("certificate has the wrong DNS names")
+	errNoPrivateKeyPEM     = errors.New("key pair is missing a private key")
+	errNoCertificatePEM    = errors.New("certificate PEM is missing")
+	errCreatedByOldCA      = errors.New("certificate was created by an older ca")
+)
+
+type certificateManager struct {
+	*x509.Certificate
```

Comment:
I would probably argue that the certificate manager is not a Certificate or a Certificate Authority, but it instead has a Certificate and Certificate Authority. I'd say some of the proof of this is in how this is used both internally and externally.

Reply:
non-blocking (since this is all hidden within this struct, but would be nice to get this correct).